### ISSUE TITLE
 CRC missmatch on QPIGS query

### ISSUE THREAD

**Issue by The-backend-dev** [b48736/axpert-monitor#6]
Hello, awesome library thanks for making it. I have encountered an issue with the QPIGS query. In my project I run it on an interval every 7 seconds, and every so often (maybe every 5 requests, and sometimes fro mthe first request) I get an error similar but not restricted to 
 CRC missmatch. Expected [ad00] but got [0bd3]

I read through the message parser and noted that you found a similar issue with QQPM sadly the fix for that did not apply here. Any help on this is appreciated. 
more info: I'm using ubuntu server 20.04 connected to my axpert inverter through micro usb, mounted on /dev/hidraw0 but I am using the HID detection and that is working fine

**Comment by b48736** [comment]
Hi, I'm glad you found the library useful.

The thing with the QOPM command is that I noticed it was consistently shifting the CRC by 4 bits, even though the payload body was correct. I assumed this was just a bug in the inverter firmware so I added a special case for it. 

I have found the serial comms on the Axpert inverters to be quite unreliable, both over USB and RS232. I also get lots of CRC errors on other commands, but they are usually off by quite a bit and not just rotated by 4 bits, so the same "fix" won't work.

With the QOPM command, the payload body is only 2 characters, so it was easy to see the data is correct and the CRC was wrong. With something like QPIGS, the body is very long and the data is rapidly changing, so it is hard to know if there is something wrong in the body, or if the body is correct and the CRC is just wrong. At some point the data should be considered unreliable and be discarded.

If you are using the library programmatically, I would recommend just putting the get call in a try/catch and ignoring the error and just retry if you get a CRC missmatch.

**Comment by The-backend-dev** [comment]
Actually, I just erased the throw error and now it works, I query every few seconds and just stream the data(and I intend to save it later) I do not particularly care about any one call. What bothered me when it was throwing the error before was that the application would then freeze, node js would not exit, nor would it retry(my code is set in an interval) I had a try catch around the block but that did not suffice. I found a way though that does not involve messing with library code. I added a process.on('uncaughtException') and that keeps my program running. So all is well, sorry for bothering you :) 
Thanks again for the library.
